src/DynamoUtilities/EnumerableExtensions.cs

```diff
@@ -467,5 +468,30 @@ where attribute.Name.Equals("value")
 
             return string.Empty;
         }
+
+        public static bool TryGetValueAs<Key, Value, ValueAs>(this IDictionary<Key, Value> dictionary, Key key, out ValueAs valueAs) where ValueAs : Value
+        {
+            if (dictionary.TryGetValue(key, out Value value))
+            {
+                valueAs = (ValueAs)value;
+                return true;
+            }
+
+            valueAs = default;
```

Comment:
Sorry, not familiar with this syntax, what is default here?

Reply:
Is assigning `null `not an option here?

Reply:
If I tweak the type constraints...then it is